src/werkzeug/routing.py

```diff
@@ -662,7 +679,15 @@ def __init__(
         self.defaults = defaults
         self.build_only = build_only
         self.alias = alias
-        if methods is None:
+        self.websocket = websocket
+        if websocket and methods is not None:
```

Comment:
Should this allows explicitly passing `methods=["GET"]`? Even though it's redundant, it's a somewhat common pattern.

Reply:
Done.